aea/crypto/fetchai.py

```diff
@@ -223,8 +222,15 @@ def validate_transaction(
 
         :return: True if the random_message is equals to tx['input']
         """
+        tx_contents = cast(TxContents, self._api.tx.contents(tx_digest))
+        is_valid = tx_contents.from_address.to_hex() == client
 
-        raise NotImplementedError
+        # TODO: Figure out the analogy of the amount and the tx_contents.fees
```

Comment:
fetchai already has the ability to recover amount: https://github.com/fetchai/ledger-api-py/blob/master/fetchai/ledger/api/tx.py#L79

we do not need to check anything regarding fees.